### PR TITLE
Expand / collapse all categories

### PR DESCRIPTION
**Pull request by joel-jeremy**
This PR is for the first item listed in #559: `Expand All / Collapse All Categories`

For the expand / collapse all categories functionality, I was choosing between having a single `Expand / collapse all categories` button or one for each: `Expand all categories` and `Collapse all categories` buttons. 

For the initial implementation, I have opted with the latter. Please let me know which one is the right way to go or if there are other suggestions and I'll just accordingly.

image

### CODE REVIEW

**Comment by j-f1** [comment]
I’m torn on what I think the right behavior should be. My first thought is that it should match the expand/collapse split transactions button. But that button has very non-intuitive behavior (it only changes state when you click it, not when you manually toggle split transactions). We also have the luxury of space for two buttons here, if we want to. What do you think, @MatissJanis?

**Comment by MatissJanis** [comment]
My framework for adding buttons is this:

Will the added feature be used by most users a lot of times per session?

Yes: it deserves prime real-estate.

No: lets hide it in a menu or other place as to not overwhelm the user with many unnecessary buttons

In this case it's a "no". So IMO hiding it in the small menu is fine. If we get a lot of people complaining that it's hard to find this - we can revisit the issue.

TLDR: I'm OK with the proposed idea from a design point of view.

**Comment by carkom** [comment]
Does or make sense to remove the word 'categories" from these 3 menu options? 

Seeing as how you are clicking a menu button right next to the word category it's not a difficult jump to assume all the menu options are for the categories so no need to write it out 3 times. Just a thought...

**Comment by joel-jeremy** [comment]
Removed the word categories in the menu options. Also thinking of renaming "Toggle hidden" to "Toggle visibility". Thoughts?

**Comment by tjfinlinson** [comment]
> Removed the word categories in the menu options. Also thinking of renaming "Toggle hidden" to "Toggle visibility". Thoughts?

I think keeping it to Toggle hidden is the right wording to use. It's a closer match to the 'hide' option in the individual category dropdown menu.

**Review by MatissJanis — APPROVED** [review]
Amazing! Thanks for your contribution @joel-jeremy !

(I'll leave the PR open for a few more days in case any other maintainers wants to review, but on my end this looks good to go)

**Review by j-f1 — APPROVED** [review]
Looks good to me!

**Comment by j-f1** [comment]
Thank you for contributing!
